### PR TITLE
mirrord-layer: Added common Result type, removed dependency of anyhow…

### PR DESCRIPTION
**Pull request by aviramha**
… crate

### CODE REVIEW

**Review by DmitryDodzin — APPROVED** [review]
Looking good.
NTH change for the macro If we want to keep it exported

File: mirrord-layer/src/macros.rs [hunk 1]
```diff
@@ -4,11 +4,12 @@ macro_rules! replace {
         let intercept = |interceptor: &mut frida_gum::interceptor::Interceptor,
                          symbol_name,
                          detour: $detour_type|
-         -> Result<$detour_type, LayerError> {
+         -> crate::error::Result<$detour_type> {
```
[the remark this answers is not in the document]
Comment (aviramha) [reply mirrord-layer/src/macros.rs:7]: Thanks!

File: mirrord-layer/src/error.rs [hunk 2]
```diff
@@ -125,6 +128,8 @@ impl<'a, T> From<std::sync::PoisonError<std::sync::MutexGuard<'a, T>>> for Layer
     }
 }
 
+pub(crate) type Result<T> = std::result::Result<T, LayerError>;
```
Comment (eyalb181) [inline mirrord-layer/src/error.rs:131]: Is it good practice to call it Result? I'm assuming most people going over the code would automatically assume it's std::Result and not realize there's a LayerError hidden there.

Comment (aviramha) [reply mirrord-layer/src/error.rs:131]: AFAIK It's more than common. https://doc.rust-lang.org/std/io/type.Result.html for example - many modules have their own `Result` aliases, and so do we.

Comment (aviramha) [reply mirrord-layer/src/error.rs:131]: P.S They can know it's not `std::Result` by seeing that only one generic is specified, and not two.

**Review by eyalb181 — APPROVED** [review]

**Pull request merged**